### PR TITLE
e2e: add cypress-terminal-report

### LINKED ISSUE #46494: add cypress-terminal-report

### PR DESCRIPTION
**Pull request by uladzimirdev**
closes https://github.com/metabase/metabase/issues/46494

### Description

add cypress terminal report

### How to verify

check a [failing job](https://github.com/metabase/metabase/actions/runs/10249928816/job/28354813499?pr=46483) and [report](https://github.com/metabase/metabase/actions/runs/10249928816/artifacts/1776624705), which can be found on the [actions page](https://github.com/metabase/metabase/actions/runs/10249928816?pr=46483)

### console output
https://github.com/user-attachments/assets/aca48e51-0898-4bea-913b-517337e93a5b

### Updated mocha report

https://github.com/user-attachments/assets/0373fc79-d8cb-4b19-9c2e-f974d18cb8ae

### CODE REVIEW

File: e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ describe("scenarios > dashboard > dashboard back navigation", () => {
 
     visitDashboard(ORDERS_DASHBOARD_ID);
     cy.wait("@dashboard");
-    cy.findByTestId("dashcard").findByText("Orders").click();
+    cy.findByTestId("dashcard").findByText("Orders123").click();
```
Comment (uladzimirdev) [inline e2e/test/scenarios/dashboard/dashboard-back-navigation.cy.spec.js:62]: revert it before merge!

File: e2e/support/cypress.js [hunk 2]
```diff
@@ -66,3 +91,26 @@ Cypress.on("window:load", window => {
...
+  cy.wait(50, { log: false }).then(() =>
+    cy.addTestContext(Cypress.TerminalReport.getLogs("txt")),
+  );
+});
+
+const options = {
+  collectTypes: [
+    "cons:log",
+    "cons:info",
+    // 'cons:warn', - disabled by intension because of noise from mbql
```
Comment (iethree) [inline e2e/support/cypress.js:106]:
```suggestion
    // 'cons:warn', - intentionally disabled because of noise from mbql
```

**Review by iethree — APPROVED** [review]
nice! ✨

File: e2e/support/cypress.js (hunk 2 above)
Comment (uladzimirdev) [reply e2e/support/cypress.js:106]: thanks, I need to start correcting comments with chatgpt

**Pull request merged**